civihr_employee_portal/js/reports.js

```diff
@@ -919,6 +919,67 @@
                         }
 
                         this.$DOM.sections.dataWrapper.fadeIn();
+                        var $DOM = this.$DOM;
+
+                        $.ajax({
+                            'type': 'GET',
+                            'url': this.getJsonUrl() + ( typeof path !== 'undefined' ? path : '' ),
+                            'success': function(data) {
+                                var mappedResults = {},
+                                    key,
+                                    countRow = $('<tr />'),
+                                    headerRow = $('<tr />'),
+                                    percentageRow = $('<tr />'),
+                                    total = 0;
+
+                                if(data.results[0] !== undefined && data.results[0].data.count !== undefined) {
+                                    for(var i in data.results) {
+                                        mappedResults[data.results[i].data.department] = +data.results[i].data.count;
+                                    }
+                                } else {
+                                    mappedResults = data.results.map(function (row) {
+                                        return {y: row.data.gender, x: 1}
+                                    }).reduce(function (accumulator, row) {
+                                        if (accumulator[row.y] === undefined) {
+                                            accumulator[row.y] = 0;
+                                        }
+
+                                        accumulator[row.y] += row.x;
+
+                                        return accumulator;
+                                    }, {});
+                                }
+
+                                for(key in mappedResults) {
+                                    total += mappedResults[key];
```

Comment:
If I understand this correctly, you can scrap the `for` loop and just write it as
```js
total = Object.keys(mappedResults).length
```